hazelcast/src/main/java/com/hazelcast/internal/dynamicclassloading/DynamicClassLoader.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.dynamicclassloading;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+public class DynamicClassLoader extends ClassLoader {
+
+    private static final ILogger LOG = Logger.getLogger(DynamicClassLoader.class);
+
+    private ClassloadingService classloadingService;
+
+    public DynamicClassLoader(ClassLoader parent) {
+        super(parent);
+    }
+
+    public void setClassloadingService(ClassloadingService service) {
+        this.classloadingService = service;
```

Comment:
Is `classloadingService` set before every thread that may call `loadClass()` starts? If not, isn't it possible `classloadingService` to be observed as null in `loadClass`?